Codecademy/gamut
# fix(ListSection): Filter out falsy children

### Overview


Updates list items to use filter out falsy list items. This fixes a bug where using an inline conditional with a list item causes extra padding.

Note: I tried updating this to use `Children.toArray(children).map` since `toArray` skips null/undefined values (TIL!) but this messed with the array keys and the only option I have for keys here is index, which we have a lint rule against.


### PR Checklist

- [x] Related to JIRA ticket: [WEB-2160]

#### Testing Instructions


See testing instructions in https://github.com/codecademy-engineering/mono/pull/628

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/33564)  | [Monolith Env](https://pr-33564-monolith.dev-eks.codecademy.com) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/628)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-628)   |




[WEB-2160]: https://codecademy.atlassian.net/browse/WEB-2160?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
